warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RemoveExtraParensVisitor.java

```diff
@@ -0,0 +1,372 @@
+import org.apache.commons.jexl2.parser.ASTEQNode;
+import org.apache.commons.jexl2.parser.ASTERNode;
+import org.apache.commons.jexl2.parser.ASTEmptyFunction;
+import org.apache.commons.jexl2.parser.ASTFalseNode;
+import org.apache.commons.jexl2.parser.ASTFloatLiteral;
+import org.apache.commons.jexl2.parser.ASTForeachStatement;
+import org.apache.commons.jexl2.parser.ASTFunctionNode;
+import org.apache.commons.jexl2.parser.ASTGENode;
+import org.apache.commons.jexl2.parser.ASTGTNode;
+import org.apache.commons.jexl2.parser.ASTIdentifier;
+import org.apache.commons.jexl2.parser.ASTIfStatement;
+import org.apache.commons.jexl2.parser.ASTIntegerLiteral;
+import org.apache.commons.jexl2.parser.ASTLENode;
+import org.apache.commons.jexl2.parser.ASTLTNode;
+import org.apache.commons.jexl2.parser.ASTMapEntry;
+import org.apache.commons.jexl2.parser.ASTMapLiteral;
+import org.apache.commons.jexl2.parser.ASTMethodNode;
+import org.apache.commons.jexl2.parser.ASTModNode;
+import org.apache.commons.jexl2.parser.ASTMulNode;
+import org.apache.commons.jexl2.parser.ASTNENode;
+import org.apache.commons.jexl2.parser.ASTNRNode;
+import org.apache.commons.jexl2.parser.ASTNullLiteral;
+import org.apache.commons.jexl2.parser.ASTNumberLiteral;
+import org.apache.commons.jexl2.parser.ASTReferenceExpression;
+import org.apache.commons.jexl2.parser.ASTReturnStatement;
+import org.apache.commons.jexl2.parser.ASTSizeFunction;
+import org.apache.commons.jexl2.parser.ASTSizeMethod;
+import org.apache.commons.jexl2.parser.ASTStringLiteral;
+import org.apache.commons.jexl2.parser.ASTTernaryNode;
+import org.apache.commons.jexl2.parser.ASTTrueNode;
+import org.apache.commons.jexl2.parser.ASTUnaryMinusNode;
+import org.apache.commons.jexl2.parser.ASTVar;
+import org.apache.commons.jexl2.parser.ASTWhileStatement;
+import org.apache.commons.jexl2.parser.JexlNode;
+import org.apache.commons.jexl2.parser.JexlNodes;
+import org.apache.commons.jexl2.parser.ParserTreeConstants;
+
+/**
+ * Input: ((FOO == 'bar')) Output: (FOO == 'bar')
+ *
+ * Preferable to {@link TreeFlatteningRebuildingVisitor#flattenAll(JexlNode)} to preserve reference expressions within marker nodes, for example bounded ranges.
+ *
+ * Note: does not operate on a copy, operates directly on the tree.
+ */
+public class RemoveExtraParensVisitor extends BaseVisitor {
+    
+    public static JexlNode remove(JexlNode node) {
+        RemoveExtraParensVisitor visitor = new RemoveExtraParensVisitor();
+        visitor.visit(node, null);
+        return node;
+    }
+    
+    /**
+     * Can remove a paren if a hierarchy exists like
+     *
+     * RefExpr -{@literal >} Ref -{@literal >} RefExpr
+     *
+     * @param node
+     *            an ASTReferenceExpression node
+     * @param data
+     *            don't need it
+     * @return
+     */
+    @Override
+    public Object visit(ASTReferenceExpression node, Object data) {
+        while (isDoubleWrapped(node)) {
```

Comment:
What if it is triple wrapped?

Reply:
This loop effectively eats up extra parens, so triple wrapped should be taken care of. This should probably check for double refs and double refExpresssions though...